packages/cmf/src/cmfConnect.js

```diff
@@ -41,12 +42,25 @@ export function getStateToProps({
 		return state.cmf.collections.get(id);
 	};
 
+	let viewProps = {};
+	if (ownProps.view) {
+		viewProps = Object.assign(
+			{},
+			state.cmf.settings.views[ownProps.view],
+		);
+		viewProps = attachRef(state, viewProps);
+		Object.keys(viewProps).forEach(
```

Comment:
Can you externalize this in an `attachRefs()` function ?
As discussed together, a better solution would be to resolve the dependency recursively only once, before the settings are injected. But this can have an impact and can be tricky. Let's keep that in mind when we have the need.